### PR TITLE
Introduce Sys Fields

### PR DESCRIPTION
**Pull request by dvaldivia**

### CODE REVIEW

File: cmd/m3/cluster-sc-sg-add.go [hunk 1]
```diff
@@ -45,17 +45,25 @@ func addStorageGroup(ctx *cli.Context) error {
...
+	if err != nil {
+		return err
+	}
+
 	// create a new storage group in the DB
-	storageGroupResult := <-cluster.AddStorageGroup(name)
+	storageGroupResult := <-cluster.AddStorageGroup(appCtx, name)
 	if storageGroupResult.Error != nil {
 		fmt.Println(storageGroupResult.Error)
+		appCtx.Rollback()
```
Comment (cesnietor) [inline cmd/m3/cluster-sc-sg-add.go:57]: [1] here there will be nothing to rollback if addStorageGroup already rolled back

File: cluster/storage-cluster.go [hunk 2]
```diff
@@ -38,35 +38,41 @@ type StorageGroupResult struct {
...
 		}
 		defer stmt.Close()
 
-		err = stmt.QueryRow(sgID, sgName).Scan(&sgNum)
+		err = stmt.QueryRow(sgID, sgName, ctx.WhoAmI).Scan(&sgNum)
 		if err != nil {
 			ch <- StorageGroupResult{
 				Error: err,
 			}
+			tx.Rollback()
```
Comment (cesnietor) [inline cluster/storage-cluster.go]: this rollback collides with the one made on `cmd/m3/cluster-sc-sg-add.go ` see comment tagged as [1]

File: cmd/m3/cluster-sc-sg-add.go (hunk 1 above)
Comment (dvaldivia) [reply cmd/m3/cluster-sc-sg-add.go:57]: It's harmless if things get rolled back twice the `ctx.Rollback()` will handle being tried to rollback a non existing transaction (as it gets closed on the previous rollback)

Comment (dvaldivia) [reply cmd/m3/cluster-sc-sg-add.go:57]: also if the internal call doesn't rollback the invoking function may decide to rollback, so it's fine to have both

**Review by cesnietor — APPROVED** [review]

**Pull request merged**